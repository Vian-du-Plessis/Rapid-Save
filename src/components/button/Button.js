import React from 'react';
import styles from './Button.module.scss';

const Button = ( props ) => {
    return (
        <button onClick={props.onClick}>
            { props.label }
        </button>
    );
};

export default Button;