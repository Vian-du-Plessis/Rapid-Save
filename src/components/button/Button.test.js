import Button from './Button';
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { act, create } from "react-test-renderer";


describe("Test the Button component", () =>{
    test("Create Rendered Button", () => {

        const testBtn = TestRenderer.create(
            <Button
                label='Add expense'
            />
        );
        const btnTestInstance = testBtn.root;
        expect(btnTestInstance.findByType(Button).props.label).toBe["Add expense"];
    });

    test("Button Matches Snapshot", async () => {
        let root;
        await act(() => {
            root = create(            
                <Button
                    label='Add expense'
                />
            );
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            root.update(
                <Button
                    label='Add expense'
                />
            );
            expect(root.toJSON()).toMatchSnapshot();
        })
    })
})
