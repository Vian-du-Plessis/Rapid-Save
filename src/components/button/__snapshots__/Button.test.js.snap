// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Button component Button Matches Snapshot 1`] = `
<button>
  Add expense
</button>
`;

exports[`Test the Button component Button Matches Snapshot 2`] = `
<button>
  Add expense
</button>
`;
